Documentation/kubernetes/install/kubespray.rst

```diff
@@ -10,7 +10,171 @@
+
+
+Installing Kubespray
+====================
+
+.. code:: bash
+
+  $ git clone git@github.com:kubernetes-incubator/kubespray.git
+
+Install dependencies from ``requirements.txt``
+
+.. code:: bash
+
+  $ cd kubespray
+  $ sudo pip install -r requirements.txt
+
+
+Infrastructure Provisioning
+===========================
+
+We will use Terraform for provisioning AWS infrastructure.
+
+-------------------------
+Configure AWS credentials
+-------------------------
+
+Export the variables for your AWS credentials 
+
+.. code:: bash
+
+  export AWS_ACCESS_KEY_ID="www"
+  export AWS_SECRET_ACCESS_KEY ="xxx"
+  export AWS_SSH_KEY_NAME="yyy"
+  export AWS_DEFAULT_REGION="zzz"
+
+-----------------------------
+Configure Terraform Variables
+-----------------------------
+
+Navigate to the kubespray cloned repo directory and rename 
+``contrib/terraform/aws/terraform.tfvars.example`` to ``terraform.tfvars``. Open the 
+file and change any defaults particularly the number of master, etcd, and worker nodes. 
+You can change the master and etcd number to 1 if using the cluster just for demo and 
+test purposes. By default, this tutorial will create:
+
+  - VPC with 2 public and private subnets
+  - Bastion Hosts and NAT Gateways in the Public Subnet
+  - Three of each (masters, etcd, and worker nodes) in the Private Subnet
+  - AWS ELB in the Public Subnet for accessing the Kubernetes API from
+    the internet
+  - Terraform scripts use ``CoreOS`` as base image.
+
+Example ``terraform.tfvars`` file:
+
+.. code:: bash
+
+  #Global Vars
+  aws_cluster_name = "kubespray"
+
+  #VPC Vars
+  aws_vpc_cidr_block = "XXX.XXX.192.0/18"
+  aws_cidr_subnets_private = ["XXX.XXX.192.0/20","XXX.XXX.208.0/20"]
+  aws_cidr_subnets_public = ["XXX.XXX.224.0/20","XXX.XXX.240.0/20"]
+
+  #Bastion Host
+  aws_bastion_size = "t2.medium"
+
+
+  #Kubernetes Cluster
+
+  aws_kube_master_num = 3
+  aws_kube_master_size = "t2.medium"
+
+  aws_etcd_num = 3
+  aws_etcd_size = "t2.medium"
+
+  aws_kube_worker_num = 3
+  aws_kube_worker_size = "t2.medium"
+
+  #Settings AWS ELB
+
+  aws_elb_api_port = 6443
+  k8s_secure_api_port = 6443
+  kube_insecure_apiserver_address = "0.0.0.0"
+
+
+-----------------------
+Apply the configuration
+-----------------------
+
+``terraform init`` to initialize the following modules
+
+  - ``module.aws-vpc``
+  - ``module.aws-elb``
+  - ``module.aws-iam``
+
+.. code:: bash
+
+  $ cd contrib/terraform/aws
+  $ terraform init
+
+Once Initialized , execute:
+
+.. code:: bash
+
+  $ terraform plan -out=aws_kubespray_plan
+
+This will generate a file ``aws_kubespray_plan`` depicting an execution
+plan of the infrastructure that will be created on AWS. To apply, execute:
+
+.. code:: bash
+
+  $ terraform init
+  $ terraform apply "aws_kubespray_plan"
+
+Terraform automatically creates an Ansible Inventory file at ``inventory/hosts``.
```

Comment:
link "Ansible Inventory file" to the documentation fo an Ansible Inventory files if such documentation exists.

Reply:
That file is just a list of the AWS resources e.g details of ec2 instances created as part of cluster.